middleware/log.go

```diff
@@ -37,7 +37,7 @@ func Log(wrappedHandler http.HandlerFunc, domain string, logOutput io.Writer) ht
 
 	return func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
-		logger = logger.WithCtx(ctx)
+		logger := mLogger.WithCtx(ctx)
```

Comment:
this is the main fix for the race identified in the github issue.   
create a new logger instance instead of mutating the main one.